### PR TITLE
Adding experimental flags for nix-command and flakes

### PR DESCRIPTION
**Pull request by Maleware**
This adds some flags to the makefile which are supposed to unblock every arm user on any system.

On Edinson I figured that this is rather a architecture thing then a MacOS problem.

### CODE REVIEW

File: template/Makefile.j2 [hunk 1]
```diff
@@ -171,7 +171,7 @@ clean: chart-clean
 regenerate-charts: chart-clean compile-chart
 
 regenerate-nix:
-	nix run -f . regenerateNixLockfiles
+	nix run --extra-experimental-features nix-command --extra-experimental-features flakes -f . regenerateNixLockfiles
```
Comment (NickLarsenNZ) [inline template/Makefile.j2:174]:
```suggestion
	nix run --extra-experimental-features "nix-command flakes" -f . regenerateNixLockfiles
```

File: template/Makefile.j2 [hunk 2]
```diff
@@ -190,7 +190,7 @@ check-kubernetes:
 
 run-dev: check-nix check-kubernetes
 	kubectl apply -f deploy/stackable-operators-ns.yaml
-	nix run -f. tilt -- up --port {[5430 + operator_index}] --namespace stackable-operators
+	nix run --extra-experimental-features nix-command --extra-experimental-features flakes -f. tilt -- up --port {[5430 + operator_index}] --namespace stackable-operators
```
Comment (NickLarsenNZ) [inline template/Makefile.j2:193]:
```suggestion
	nix run --extra-experimental-features "nix-command flakes" -f. tilt -- up --port {[5430 + operator_index}] --namespace stackable-operators
```

File: template/Makefile.j2 [hunk 3]
```diff
@@ -190,7 +190,7 @@ check-kubernetes:
...
 
 stop-dev: check-nix check-kubernetes
-	nix run -f. tilt -- down
+	nix run --extra-experimental-features nix-command --extra-experimental-features flakes -f. tilt -- down
```
Comment (NickLarsenNZ) [inline template/Makefile.j2:196]:
```suggestion
	nix run --extra-experimental-features "nix-command flakes" -f. tilt -- down
```

**Review by NickLarsenNZ — APPROVED** [review]
Feel free to use the suggestions I left

File: template/Makefile.j2 (hunk 2 above)
Comment (Maleware) [reply template/Makefile.j2]: thanks, done :)

File: template/Makefile.j2 (hunk 3 above)
Comment (Maleware) [reply template/Makefile.j2]: Same, Thanks :)

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

File: template/Makefile.j2 (hunk 3 above)
Comment (NickLarsenNZ) [reply template/Makefile.j2]: adea4cc

File: template/Makefile.j2 (hunk 2 above)
Comment (NickLarsenNZ) [reply template/Makefile.j2]: adea4cc

File: template/Makefile.j2 (hunk 1 above)
Comment (NickLarsenNZ) [reply template/Makefile.j2]: adea4cc

**Pull request merged**